lib/infra/infra-stack.ts

```diff
@@ -79,35 +82,71 @@ export class InfraStack extends Stack {
     const ec2InstanceType = (props.cpuType === AmazonLinuxCpuType.X86_64)
       ? InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE) : InstanceType.of(InstanceClass.C6G, InstanceSize.XLARGE);
 
+    const alb = new NetworkLoadBalancer(this, 'publicNlb', {
+      vpc: props.vpc,
+      internetFacing: true,
+      crossZoneEnabled: true,
+    });
+
+    if (!props.securityDisabled && !props.minDistribution) {
+      opensearchListener = alb.addListener('opensearch', {
+        port: 443,
+        protocol: Protocol.TCP,
+      });
+    } else {
+      opensearchListener = alb.addListener('opensearch', {
+        port: 80,
+        protocol: Protocol.TCP,
+      });
+    }
+
+    if (props.dashboardsUrl !== 'undefined') {
+      dashboardsListener = alb.addListener('dashboards', {
+        port: 8443,
+        protocol: Protocol.TCP,
+      });
+    }
+
     if (props.singleNodeCluster) {
       console.log('Single node value is true, creating single node configurations');
-      const singleDataNodeAsg = new AutoScalingGroup(this, 'dataNodeAsg', {
+      singleNodeInstance = new Instance(this, 'single-node-instance', {
         vpc: props.vpc,
         instanceType: ec2InstanceType,
         machineImage: MachineImage.latestAmazonLinux({
           generation: AmazonLinuxGeneration.AMAZON_LINUX_2,
           cpuType: props.cpuType,
         }),
         role: instanceRole,
-        maxCapacity: 1,
-        minCapacity: 1,
-        desiredCapacity: 1,
         vpcSubnets: {
           subnetType: SubnetType.PRIVATE_WITH_EGRESS,
         },
         securityGroup: props.securityGroup,
         blockDevices: [{
           deviceName: '/dev/xvda',
-          volume: BlockDeviceVolume.ebs(50, { deleteOnTermination: true }),
+          volume: BlockDeviceVolume.ebs(100, { deleteOnTermination: true }),
```

Comment:
Do we want to make this value optional? Wondering if anyone wanting to use large data set might need higher volume. Default to 100?

Reply:
I was expecting this change from a contributor but I think it makes more sense to just add it as well. Will update the PR.